### PR TITLE
feature-cloudapi/PW-396

### PR DESCRIPTION
**Pull request by maassenbas**
- Updated metadata import file 
  - Configurable POIID & X-API-KEY in Custom Preferences
  - Payment method & Payment Processor
  - Payment service 
- Authorise call Sync flow
- Redirect to result page

### CODE REVIEW

File: cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            customerID = profile.getCustomerNo();
+        }
+
+        var shopperObject = AdyenHelper.createShopperObject({
+            Customer: customer,
+            Basket: order
+        });
+
+        var d = new Date();
+	    var dateString =  d.getTime().toString();
```
Comment (rikterbeek) [inline cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds]: fix indenting

File: cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+						         "SaleID":"SalesforceCommerceCloud",
+						         "POIID": poiid
+						      },
+						      "PaymentRequest":{  
+						         "SaleData":{  
+						            "SaleTransactionID":{  
+						               "TransactionID": order.getOrderNo(),
+						               "TimeStamp": new Date()
+						            },
+						            "SaleReferenceID":"TestSFCCpos"
```
Comment (rikterbeek) [inline cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds]: Let's not call it test could lead to confusion. SalesforceCommerceCloudPOS

File: cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+						         },
+						         "PaymentTransaction":{  
+						            "AmountsReq":{  
+						               "Currency": "EUR", //args.Amount.currencyCode,
```
Comment (rikterbeek) [inline cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds]: this is fixed let's use currency of order.

File: cartridges/int_adyen/cartridge/templates/default/pos.isml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<iscontent type="text/html" charset="UTF-8" compact="true"/>
+<isinclude template="util/modules"/>
+
+<isset name="brandCode" value="" scope="session"/>
+	<ul id="type">
+			<li class="form-row row-POS">
+					<input type="radio" class="input-radio" name="brandCode" value="AdyenPOS"/>
```
Comment (rikterbeek) [inline cartridges/int_adyen/cartridge/templates/default/pos.isml]: check indenting.

File: cartridges/int_adyen_controllers/cartridge/scripts/payment/processor/Adyen_POS.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+var Transaction = require('dw/system/Transaction');
+
+/* Script Modules */
+var app = require('app_storefront_controllers/cartridge/scripts/app');
+
+/**
+ * Creates a Adyen payment instrument for the given basket
+ */
+function handle(args) {
+	var	adyenRemovePreviousPI = require('int_adyen/cartridge/scripts/adyenRemovePreviousPI'),
```
Comment (rikterbeek) [inline cartridges/int_adyen_controllers/cartridge/scripts/payment/processor/Adyen_POS.js]: a lot of spaces after var.

File: cartridges/int_adyen_controllers/cartridge/scripts/payment/processor/Adyen_POS.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+ * Authorizes a payment using a POS terminal. 
+ * The payment is authorized by using the Adyen_POS processor only and setting the order no as the transaction ID. 
+ */
+function authorize(args) { 
+	var AdyenHelper = require('int_adyen/cartridge/scripts/util/AdyenHelper');
+	
+	var order = args.Order;
+    var orderNo = args.OrderNo;
```
Comment (rikterbeek) [inline cartridges/int_adyen_controllers/cartridge/scripts/payment/processor/Adyen_POS.js]: check indenting.

File: cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        var shopperObject = AdyenHelper.createShopperObject({
+            Customer: customer,
+            Basket: order
+        });
+
+        var date = new Date();
+	    var dateString = date.getTime().toString();
+    	var serviceId = dateString.substr(dateString.length - 10);
+	    var poiid = AdyenHelper.getAdyenPOIID();
+	    var myAmount = AdyenHelper.getCurrencyValueForApi(args.Amount) / 100;
```
Comment (lancergr) [inline cartridges/int_adyen/cartridge/scripts/adyenPosVerification.ds]: what about using just args.Amount?

**Pull request merged**